projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget.model.ts

```diff
@@ -165,15 +183,102 @@ export class TableWidgetModel {
   })
   public pageable?: boolean = true;
 
+  @ModelProperty({
+    key: 'fetchEditableColumns',
+    displayName: 'Query for additional columns not provided',
+    type: BOOLEAN_PROPERTY.type
+  })
+  public fetchEditableColumns?: boolean = false;
+
   @ModelInject(MODEL_API)
   private readonly api!: ModelApi;
 
+  @ModelInject(MetadataService)
+  private readonly metadataService!: MetadataService;
+
+  @ModelInject(FilterBuilderLookupService)
+  private readonly filterBuilderLookupService!: FilterBuilderLookupService;
+
   public getData(): Observable<TableDataSource<TableRow>> {
     return this.api.getData<TableDataSource<TableRow>>();
   }
 
-  public getColumns(): SpecificationBackedTableColumnDef[] {
-    return this.columns.map(column => column.asTableColumnDef());
+  public getColumns(scope?: string): Observable<SpecificationBackedTableColumnDef[]> {
+    const modelColumns = this.columns.map(column => column.asTableColumnDef());
+
+    if (scope === undefined || !this.fetchEditableColumns) {
+      return of(modelColumns);
+    }
+
+    return this.metadataService.getSelectionAttributes(scope).pipe(
+      map(attributes => [
+        ...modelColumns,
+        ...attributes
+          .filter(attribute => this.filterNotModelColumnAttribute(attribute, modelColumns))
+          .filter(attribute => this.filterEditableAttribute(attribute))
+          .map(attribute => this.mapAttributeToColumnConfig(attribute))
```

Comment:
Hmm. This is the first I'm hearing this. Might need some more explanation.

So we will get an AttributeMetadata object back and it will have that we want to split into multiple columns?

Reply:
So take a metric column like response time - what column do we make out of that? P99 response time? avg response time? max? Each metric could correspond to N different columns (where N is the # of supported aggregations). This is what I meant but probably forgot to say explicitly when I had expressed some concern that generating columns may produce a huge list of options, many of which may just be noise.

Reply:
Ah, gotcha. We'ree not adding aggregations to metric attributes yet, but it makes sense to eventually.

Reply:
By not adding aggregations to metrics, you mean we're not using any attributes here with `onlySupportsAggregation` true in our fancy new lingo? I think I missed the logic in here for that, or is that still a WIP?

Reply:
Correct. That boolean will be checked in `filterEditableAttribute` or whatever we name it. I can replace the switch statement.